Enforce that positive integers must be positive

jwj61 opened the pull request:
Addresses issue #4436 by enforcing that the invariant factors of the class group need to be non-zero.

https://beta.lmfdb.org/NumberField/?hst=List&galois_group=D10&class_number=5&class_group=%5B0%2C1%2C5%5D&search_type=List
http://127.0.0.1:37777/NumberField/?hst=List&galois_group=D10&class_number=5&class_group=%5B0%2C1%2C5%5D&search_type=List

roed314 commented:
There's a test failure: `self.check_args('/NumberField/?start=0&degree=6&signature=%5B0%2C3%5D&count=100', '6.0.61131.1')`

jwj61 commented:
I think tests will pass this time.  I added a flag to the parsing function as to whether to allow 0 to be a positive integer.  I also grepped for other places this function was called.  Most did not want to allow 0 (usually torsion subgroups), but at least one other did want to allow 0 (higher genus curves), so I adjusted that too.
